connect/src/com/telenor/connect/sms/SmsPinParseUtil.java

```diff
@@ -7,6 +7,8 @@
 
 public class SmsPinParseUtil {
 
+    private static final Pattern PIN_PATTERN = Pattern.compile("<#>.*([0-9]{4}).*.{11}");
```

Comment:
Is this pattern the entire message, so should have anchoring with ^ and $?

Reply:
Kinda/perhaps. I know some BUs prefix their sms with the cost of the sms. It shouldn't be picked up by Android then, as it doesn't match the requirements, but I didn't want to explicitly block it, in case it did work. Don't know if there is any point to it.